src/pybind/mgr/dashboard/services/auth.py

```diff
@@ -277,3 +288,64 @@ def _check_authorization(self, username):
         if not AuthManager.authorize(username, sec_scope, sec_perms):
             raise cherrypy.HTTPError(403, "You don't have permissions to "
                                           "access that resource")
+
+class Oauth2JWTManager(object):
+    _token = ''
+    _token_payload = {}
+    logger = logging.getLogger('oauth2jwt')
+
+
+    @classmethod
+    def set_token(cls, token):
+        cls._token = token
+        cls.set_token_payload()
+        JwtManager.oauth2_token = True
```

Comment:
sounds like this is repeated: you area already setting this variable inside the method `set_token_payload` ... and logic is different because inside the method you are setting it conditionally!

Reply:
updated it